sensision/src/main/java/io/warp10/sensision/OpenMetrics.java

```diff
@@ -0,0 +1,189 @@
+    //
+
+    StringBuilder sb = null;
+
+    for (int i = 0; i < name.length(); i++) {
+      if (name.charAt(i) >= 'a' && name.charAt(i) <= 'z') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (name.charAt(i) >= 'A' && name.charAt(i) <= 'Z') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (name.charAt(i) >= '0' && name.charAt(i) <= '9') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if ('_' == name.charAt(i) || ':' == name.charAt(i)) {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (null == sb) {
+        sb = new StringBuilder(name.length());
+        sb.append(name, 0, i);
+      }
+      // Replace forbidden char by '_'
+      sb.append("_");
+    }
+
+    //
+    // Check the first char
+    //
+
+    if (null == sb) {
+      if (name.charAt(0) >= '0' && name.charAt(0) <= '9') {
+        return "_" + name;
+      } else {
+        return name;
+      }
+    } else {
+      if (sb.charAt(0) >= '0' && sb.charAt(0) <= '9') {
+        return "_" + sb.toString();
+      } else {
+        return sb.toString();
+      }
+    }
+  }
+
+  private static final String sanitizeLabelName(String name) {
+    //
+    // Label names in OpenMetrics can only start by a letter or '_' and can only
+    // contain letters, digits and '_'
+    //
+
+    StringBuilder sb = null;
+
+    for (int i = 0; i < name.length(); i++) {
+      if (name.charAt(i) >= 'a' && name.charAt(i) <= 'z') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (name.charAt(i) >= 'A' && name.charAt(i) <= 'Z') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (name.charAt(i) >= '0' && name.charAt(i) <= '9') {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if ('_' == name.charAt(i)) {
+        if (null != sb) {
+          sb.append(name.charAt(i));
+        }
+        continue;
+      }
+      if (null == sb) {
+        sb = new StringBuilder(name.length());
+        sb.append(name, 0, i);
+      }
+      // Replace forbidden char by '_'
+      sb.append("_");
+    }
+
+    //
+    // Check the first char
+    //
+
+    if (null == sb) {
+      if (name.charAt(0) >= '0' && name.charAt(0) <= '9') {
+        return "_" + name;
+      } else {
+        return name;
+      }
+    } else {
+      if (sb.charAt(0) >= '0' && sb.charAt(0) <= '9') {
+        return "_" + sb.toString();
+      } else {
+        return sb.toString();
+      }
+    }
+  }
+
+  private static String sanitizeLabelValue(String value) {
+    //
+    // Label values must have \n, \ and " escaped by \
+    //
+
+    value = value.replaceAll("\\", "\\\\");
```

Comment:
This will raise an exception:
`java.util.regex.PatternSyntaxException: Unexpected internal error near index 1
\`